### PR TITLE
Update Jobs Page

### PR DESCRIPTION
**Pull request by slightlyoffbeat**
This addresses #8 

This will

1. update the EN jobs page by removing the list of open positions and replacing with a link to the open positions on the mozilla careers page.

2. Remove all non-EN jobs pages because they are not localized.

### CODE REVIEW

**Comment by slightlyoffbeat** [comment]
There is currently a discussion around redirects. This should not be merged until we know how redirects will be handled.

File: content/en/jobs/index.html [hunk 1]
```diff
@@ -258,24 +258,7 @@ <h1>Join the Pocket Team</h1>
...
-                    <li>
-                      <a
-                        href="https://boards.greenhouse.io/pocketco/jobs/2436558"
-                        target="_blank"
-                        rel="noopener noreferrer"
-                        >Staff Software Engineer - iOS</a
-                      >
-                    </li>
-                  </ul>
+                  <p><a href="https://careers.mozilla.org/listings/?team=Pocket" target="_blank" rel="noopener noreferrer">Click here</a> to see a list of positions we are hiring for.</p>
```
Comment (craigcook) [inline content/en/jobs/index.html:261]: Can we avoid "click here" as link text?
```suggestion
                  <p><a href="https://careers.mozilla.org/listings/?team=Pocket" target="_blank" rel="noopener noreferrer">See our latest job openings</a></p>
```

**Review by craigcook — CHANGES_REQUESTED** [review]

**Review by craigcook — APPROVED** [review]
🧑‍🏭

**Comment by slightlyoffbeat** [comment]
@craigcook I like it, and went ahead and accepted your suggestion.

**Pull request merged**